app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def index; end
+
+  def edit; end
+
+  def update
+    current_user.update(user_params)
+    if current_user.valid?
+      flash[:success] = "Successfully updated your profile."
+    else
+      flash[:error] = current_user.errors.messages.to_s
```

Comment:
I don't think this is quite sufficient for composing a sensible error message. When I intentionally triggered an error, this is how it displayed:

<img width="507" alt="error" src="https://user-images.githubusercontent.com/4731/76588488-2886c300-64b5-11ea-9783-1ca4a71614e3.png">

And two:

<img width="572" alt="error-two" src="https://user-images.githubusercontent.com/4731/76588516-3b999300-64b5-11ea-9cd8-bf0df21b850a.png">

Reply:
Ah yes, that is a problem I forgot to completely remedy. There is a proper rails way that i'll implement today. Error handling on forms is like pulling teeth for me.

Reply:
Fixed